content/qemu/index.md

```diff
@@ -3,52 +3,109 @@ title = 'QEMU'
-apt-get update ; apt-get install qemu # install qemu
-modprobe kqemu                        # load the kernel module
-mknod /dev/kqemu c 250 0              # create the device
-chmod 666 /dev/kqemu /dev/net/tun     # adjust permissions
+apt update && apt install qemu-system        # install qemu
+modprobe kvm                                 # load the kernel module
+adduser $USER kvm                            # add your user to kvm group
+# now login again or try: newgrp kvm
 </pre>
 
-To see if kqemu is enabled and working correctly, use the QEMU monitor command (press ctrl-alt-2): 'info kqemu'
+To see if kvm is enabled and working correctly, use the QEMU monitor command (press ctrl-alt-2): 'info kvm'
 
-## Problems with Qemu
+## Usage examples
 
-### Black / blank screen
+In all examples below you can add `-nographic` to get the first serial console on your terminal.
+This might work better on arm64, where most software defaults to supporting serial console, and less good on amd64.
 
-You get a black/blank screen when booting grml in qemu? Try booting with &quot;grml vga=normal&quot;.
+The example use `-m 2048` for 2GB (2048MB) of virtual memory (RAM).
+They do not pass any CPU core counts, so you get one virtualized CPU core.
 
-## Running QEMU with grml on Linux
+### Boot Grml ISO on arm64 (aarch64)
 
-Install qemu (apt-get install qemu) and start it e.g. via:
+This assumes the firmware shall be UEFI EDK2 ("TianoCore").
+This is the best supported QEMU target on arm64.
+
+This also assumes that your host is actually arm64 and has KVM support.
 
 <pre class="rahmen">
-qemu -m 256 -cdrom /dev/hdc          # if running from CD-ROM
-qemu -m 256 -cdrom /path/to/grml.iso # if running from harddisk
+apt update && apt install qemu-system qemu-system-gui qemu-efi-aarch64
+cp /usr/share/AAVMF/AAVMF_VARS.fd efi_vars.fd
+qemu-system-aarch64 \
+    -M virt -cpu host -machine gic-version=host \
+    -enable-kvm \
+    -m 2048 \
+    -drive if=pflash,format=raw,unit=0,file.filename=/usr/share/AAVMF/AAVMF_CODE.no-secboot.fd,file.locking=off,readonly=on \
+    -drive if=pflash,unit=1,file=efi_vars.fd \
+    -monitor unix:qemu-monitor-socket,server,nowait \
+    -cdrom ~/grml-full-{{< param current_release.version >}}-arm64.iso
 </pre>
 
-## Running QEMU with grml on Windows
+### Boot Grml ISO on amd64 (64-bit PC), UEFI
+
+This assumes the firmware shall be UEFI EDK2 ("TianoCore").
+This also assumes that your host is amd64 and has KVM support.
+
+<pre class="rahmen">
+apt update && apt install qemu-system qemu-system-gui ovmf
+cp /usr/share/OVMF/OVMF_VARS_4M.fd efi_vars.fd
+qemu-system-aarch64 \
+    -M q35 -cpu host \
+    -enable-kvm \
+    -m 2048 \
+    -drive if=pflash,format=raw,unit=0,file.filename=/usr/share/OVMF/OVMF_CODE_4M.fd,file.locking=off,readonly=on \
+    -drive if=pflash,unit=1,file="efi_vars.fd \
+    -monitor unix:qemu-monitor-socket,server,nowait \
+    -cdrom ~/grml-full-{{< param current_release.version >}}-amd64.iso
```

Comment:
I guess this (and below) should be `qemu-system-x86_64`.

Maybe tell the user why `q35` (the default (pc-i440fx-7.2), works fine too)?

I use `-monitor stdio` when not using `-nographics` or `-serial`. Any benefits for using the unix socket?

Reply:
Unix socket is mostly so its the same as on arm64.

Reply:
i've added `-monitor stdio` to the x86 examples, and merged the monitor section into the arm64 example